src/OpenTelemetry.Instrumentation.GrpcNetClient/Implementation/GrpcClientDiagnosticListener.cs

```diff
@@ -135,16 +142,33 @@ public void OnStartActivity(Activity activity, object payload)
                 }
 
                 var uriHostNameType = Uri.CheckHostName(request.RequestUri.Host);
-                if (uriHostNameType == UriHostNameType.IPv4 || uriHostNameType == UriHostNameType.IPv6)
+                if (this.emitOldAttributes)
                 {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerIp, request.RequestUri.Host);
+                    if (uriHostNameType == UriHostNameType.IPv4 || uriHostNameType == UriHostNameType.IPv6)
+                    {
+                        activity.SetTag(SemanticConventions.AttributeNetPeerIp, request.RequestUri.Host);
+                    }
+                    else
+                    {
+                        activity.SetTag(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host);
+                    }
+
+                    activity.SetTag(SemanticConventions.AttributeNetPeerPort, request.RequestUri.Port);
                 }
-                else
+
+                if (this.emitNewAttributes)
                 {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host);
-                }
+                    if (uriHostNameType == UriHostNameType.IPv4 || uriHostNameType == UriHostNameType.IPv6)
+                    {
+                        activity.SetTag(SemanticConventions.AttributeServerSocketAddress, request.RequestUri.Host);
```

Comment:
It looks like we missed out on updating `net.peer.ip` to `server.socket.address` for AspNetCore instrumentation: https://github.com/open-telemetry/opentelemetry-dotnet/blob/18f79788de0696654c053acedee5d991a6c026f3/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L493-L497

Reply:
It will be `client.address` actually for asp.net core.

Reply:
I'm working on a new PR to fix this in the AspNetCore library.